packages/i18n/src/index.ts

```diff
@@ -186,39 +190,39 @@ export const resolveTemplate: TemplateResolver = (string: string, args?: BaseTem
 /**
  * Template resolver that does nothing. It's used as a fallback when no template resolver is provided.
  */
-export const identityResolveTemplate = (v => v) as TemplateResolver;
+export const identityResolveTemplate = (v => v) as TemplateResolver<any, any>;
 
-export type Resolved<T, O> = T extends (...args: any[]) => infer R ? R : T extends O ? O : T;
+export type Resolved<Dict, Out> = Dict extends (...args: any[]) => infer R ? R : Out;
 
-export type ResolveArgs<T, O> = T extends (...args: infer A) => any
+export type ResolveArgs<Dict, Out, Args> = Dict extends (...args: infer A) => any
```

Comment:
This should have a default value for `Args`, right?

Reply:
I can definetly add it yes